### PR TITLE
Feat: Added Conflict Reasons

### LINKED ISSUE #708: Add ConflictReasons strings to python (and rust)
## Description

https://github.com/iotaledger/iota-sdk/pull/639 added 2 files that we need to add to python and investigate how to do in rust:
- [bindings/nodejs/lib/types/models/conflict-reason.ts](https://github.com/iotaledger/iota-sdk/pull/639/files/57fbc09cf254aa3137d12ccd84d549a71948a05e#diff-5042ca1c916f2ed87dd6d38fd9f0af535d99c36e6d31104c36fbd9e954578d8e)

## Motivation

Feature parity

## Open questions (optional)

Do we actually want this?

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

fixes #708 

Open question:
Rust; do we want it in `fmt::Display` or in a `.reason_str()` method?

### CODE REVIEW

**Review by thibault-martinez — COMMENTED** [review]
May deserves a small entry in the changelogs

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI

**Review by Thoralf-M — DISMISSED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
CI

File: bindings/python/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+
+## 1.0.1 - 2023-MM-DD
```
Comment (thibault-martinez) [inline bindings/python/CHANGELOG.md]:
```suggestion
## 1.1.0 - 2023-MM-DD
```

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**